site/_posts/2022-06-28-remix-demo.md

```diff
@@ -0,0 +1,152 @@
+  // If the user is already authenticated redirect to /dashboard directly
+  return await authenticator.authenticate("FusionAuth", request, {
+    successRedirect: "/dashboard",
+    failureRedirect: "/error",
+  },
+  );
+};
+```
+
+This is all the app code you need to protect any route. As you can see, you will merely be trying the `authenticator.authenticate` method, and redirecting to either the `/dashboard` or the `/error` route on success or failure.
+
+What is this authenticator thing all about? If you open the `/app/auth.server` file, you should see something like this (slightly simplified here):
+
+```bash
+import { Authenticator } from "remix-auth";
+import { sessionStorage } from "~/session.server";
+import { OAuth2Strategy } from "remix-auth-oauth2";
+
+// Create an instance of the authenticator, pass a generic with what
+// strategies will return and will store in the session
+export let authenticator = new Authenticator<User>(sessionStorage);
+
+authenticator.use(
+    new OAuth2Strategy(
+        {
+            authorizationURL: `${process.env.AUTH_URL}/authorize`,
+            tokenURL: `${process.env.AUTH_URL}/token`,
+            clientID: process.env.CLIENT_ID || "",
+            clientSecret: process.env.CLIENT_SECRET || "",
+            callbackURL: process.env.AUTH_CALLBACK_URL || "",
+
+        },
+        async ({ accessToken, refreshToken, extraParams, profile, context }) => {
+            // This function is MANDATORY for the system to work, and would be the
+            // main cause of being redirected to the /error route
+            console.log("Verified by FusionAuth!")
+        }
+    ),
+    // this is optional, but if you setup more than one OAuth2 instance you will
+    // need to set a custom name to each one
+    "FusionAuth"
+);
+```
+
+This code leans very heavily on @sergiodxa's OAuth work for Remix, the npm packages `remix-auth` and `remix-auth-oauth2`. It will handle all the tedious boilerplate necessary to negotiate between Remix apps and an OAuth2 authentication service like FusionAuth. As you can see, we just send some secret data -- and remember, this is all sent from the server side of Remix, not from client-side JavaScript! -- which helps tell FusionAuth what app you are trying to log in from, and where you want to go afterwards.
+
+If you are not logged in, our Remix `/login` route will redirect you to the FusionAuth login -- which you can [customize easily](/docs/v1/tech/themes/) with your own CSS! -- which looks like this out of the box:
+
+{% include _image.liquid src="/assets/img/blogs/connecting-fusionauth-remix/dinesh-login.png" alt="FusionAuth login screen" class="img-fluid" figure=true %}
+
+and if you loaded the Kickstart data as directed you should now be able to log in as `dinesh@fusionauth.io` with password `password`. Any specifically login-related errors, like forgetting your password or not being registered, will be handled here.
```

Comment:
Where did we discuss Kickstart? Is that missing from the guide?

Reply:
Removed.